app/controllers/files_controller.rb

```diff
@@ -0,0 +1,20 @@
+class FilesController < ApplicationController
+  def upload
+    if File.file?(Upload.base_upload_dir.join(params[:path]))
+      case File.extname(params[:path]).downcase
+      when ".jpg", ".jpeg", ".png", ".gif"
+        disp = "inline"
+      else
+        disp = "attachment"
+      end
+      mime = mime_type(params[:path])
+      
+      send_file Upload.base_upload_dir.join(params[:path]).to_s,
```

Comment:
We could keep /public and then tell nginx that it should serve the files there as application/octet-stream, content-disposition: attach.

There are probably browser edge cases though, and I'm not sure what happens if you <img src="..."> to an attachment.